exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SetOperations.kt

```diff
@@ -26,7 +26,8 @@ sealed class SetOperation(
         is Query -> {
             val newSlice = _firstStatement.set.fields.mapIndexed { index, expression ->
                 when (expression) {
-                    is Column<*>, is ExpressionAlias<*> -> expression
+                    is Column<*>, is ExpressionWithColumnTypeAlias<*>, is ExpressionAlias<*> -> expression
```

Comment:
Looks like here it could be also replaced with `is Column<*>, is IExpressionAlias<*> -> expression`